azext_edge/tests/edge/orchestration/resources/connector/opcua/test_opcua_certs_unit.py

```diff
@@ -0,0 +1,913 @@
+                },
+            },
+            get_mock_spc_record(spc_name=OPCUA_TRUST_LIST_SECRET_SYNC_NAME, resource_group_name="mock-rg"),
+            get_mock_secretsync_record(
+                secretsync_name=OPCUA_TRUST_LIST_SECRET_SYNC_NAME,
+                resource_group_name="mock-rg",
+                objects=[
+                    {
+                        "sourcePath": "secret1",
+                        "targetKey": "certificate.der",
+                    }
+                ],
+            ),
+            "/fake/path/certificate.der",
+            "new-secret",
+            get_mock_secretsync_record(
+                secretsync_name=OPCUA_TRUST_LIST_SECRET_SYNC_NAME,
+                resource_group_name="mock-rg",
+                objects="new-secret",
+            ),
+        ),
+    ],
+)
+def test_trust_add(
+    mocker,
+    mocked_cmd,
+    mocked_logger: Mock,
+    mocked_sleep: Mock,
+    expected_resources_map,
+    trust_list_spc,
+    trust_list_secretsync,
+    file_name,
+    file_content,
+    secret_name,
+    expected_secret_sync,
+    mocked_responses: responses,
+):
+
+    mocker.patch(
+        "azext_edge.edge.providers.orchestration.resources.connector.opcua.certs.OpcUACerts._get_cl_resources",
+        return_value=expected_resources_map["resources"],
+    )
+    mocker.patch(
+        "azext_edge.edge.providers.orchestration.resources.connector.opcua.certs.read_file_content",
+        return_value=file_content,
+    )
+    instance_name = generate_random_string()
+    rg_name = "mock-rg"
+
+    if expected_resources_map["resources"]:
+        # get default spc
+        mocked_responses.add(
+            method=responses.GET,
+            url=get_spc_endpoint(spc_name="default-spc", resource_group_name=rg_name),
+            json=expected_resources_map["resources"][0],
+            status=200,
+            content_type="application/json",
+        )
+
+    if trust_list_spc:
+        _setup_mock_common_responses(
+            mocked_responses=mocked_responses,
+            spc=trust_list_spc,
+            secretsync=trust_list_secretsync,
+            opcua_secretsync_name=OPCUA_TRUST_LIST_SECRET_SYNC_NAME,
+            rg_name=rg_name,
+            secret_name=secret_name,
+        )
+
+        matched_target_key = False
+        mapping = trust_list_secretsync["properties"]["objectSecretMapping"]
+
+        if mapping:
+            matched_target_key = mapping[0]["targetKey"] == os.path.basename(file_name)
+
+        if not matched_target_key:
+            # set opcua secretsync
+            mocked_responses.add(
+                method=responses.PUT,
+                url=get_secretsync_endpoint(
+                    secretsync_name=OPCUA_TRUST_LIST_SECRET_SYNC_NAME, resource_group_name=rg_name
+                ),
+                json=expected_secret_sync,
+                status=200,
+                content_type="application/json",
+            )
+
+    result = None
+
+    try:
```

Comment:
with pytest.raises exists. You should use that (and divide this test into test_trust_add and test_trust_add_error)